common/ayon_common/distribution/addon_distribution.py

```diff
@@ -312,185 +361,694 @@ def _try_convert_to_server_source(addon, source):
+
+    @property
+    def dependency_package(self):
+        if self._dependency_package == -1:
+            self._dependency_package = get_dependency_package()
+        return self._dependency_package
+
+    def _prepare_current_addons_state(self):
+        addons_metadata = self.get_addons_metadata()
+        output = {}
+        for full_name, addon_info in self.addons_info.items():
+            if not addon_info.require_distribution:
+                continue
+            addon_dest = os.path.join(self._addons_dirpath, full_name)
+            self.log.debug(f"Checking {full_name} in {addon_dest}")
+            addon_in_metadata = (
+                addon_info.name in addons_metadata
+                and addon_info.version in addons_metadata[addon_info.name]
+            )
+            if addon_in_metadata and os.path.isdir(addon_dest):
+                self.log.debug(
+                    f"Addon version folder {addon_dest} already exists."
+                )
+                state = UpdateState.UPDATED
+
+            else:
+                state = UpdateState.OUTDATED
+
+            downloader_data = {
+                "type": "addon",
+                "name": addon_info.name,
+                "version": addon_info.version
+            }
+            sources = []
+            for source in addon_info.sources:
+                if source.type == UrlType.HTTP.value:
+                    source = _try_convert_to_server_source(addon_info, source)
+                sources.append(source)
+
+            output[full_name] = DistributionItem(
+                state,
+                addon_dest,
+                addon_dest,
+                addon_info.hash,
+                self._dist_factory,
+                sources,
+                downloader_data,
+                full_name,
+                self.log
+            )
+        return output
+
+    def _preapre_dependency_progress(self):
+        package = self.dependency_package
+        if package is None or not package.require_distribution:
+            return None
+
+        metadata = self.get_dependency_metadata()
+        downloader_data = {
+            "type": "dependency_package",
+            "name": package.name,
+            "platform": package.platform
+        }
+        zip_dir = package_dir = os.path.join(
+            self._dependency_dirpath, package.name
+        )
+        self.log.debug(f"Checking {package.name} in {package_dir}")
+
+        if not os.path.isdir(package_dir) or package.name not in metadata:
+            state = UpdateState.OUTDATED
+        else:
+            state = UpdateState.UPDATED
+
+        return DistributionItem(
+            state,
+            zip_dir,
+            package_dir,
+            package.checksum,
+            self._dist_factory,
+            package.sources,
+            downloader_data,
+            package.name,
+            self.log,
+        )
+
+    def get_addons_progress(self):
+        if self._addons_progress is None:
```

Comment:
There is a lot of docstrings missing, that this method should be doing? I see some discrepancy between `progress` and `state` (`_prepare_current_addons_state`). Why was 'progress' is preferable from 'state', 'status'?
What it is supposed to be returning? There is something called 'DistributeTransferProgress` so I would expect that `get_addons_progress` returns these, but it is is actually returning dictionary of something called `DistributionItem`.

Reply:
Isn't this something like a queue of addons which should be downloaded?

Reply:
Made this in rush and didn't think about it, will look into that.